### PR TITLE
Split content of StringFunctions.h into binary, string, and url functions

### PR DESCRIPTION
**Pull request by isadikov**
This PR moves binary-related functions to `BinaryFunctions.h` and url-related functions to `URLFunctions.h` from `StringFunctions.h` according to the documentation, i.e. `binary.rst`, `string.rst`, and `url.rst`. Tests are also split in the same way - there is a total of original 56 tests split into 32 tests in StringFunctionsTest.cpp, 22 tests in BinaryFunctionsTest.cpp, and 2 tests in URLFunctionsTest.cpp.

This is a follow-up for https://github.com/facebookincubator/velox/pull/4281 where this separation was requested.

### CODE REVIEW

**Comment by isadikov** [comment]
cc @aditi-pandit @mbasmanova I thought maybe we could just merge this PR, it is a simple split with all of the tests preserved (spread across several files). Thanks.

File: velox/functions/prestosql/StringFunctions.h [hunk 1]
```diff
@@ -52,255 +52,6 @@ struct CodePointFunction {
   }
 };
 
-/// crc32(varbinary) → bigint
-/// Return an int64_t checksum calculated using the crc32 method in zlib.
-template <typename T>
```
Comment (mbasmanova) [inline velox/functions/prestosql/StringFunctions.h:57]: Can we maybe remove some of the #includes now?

File: velox/functions/prestosql/URLFunctions.h [hunk 2]
```diff
@@ -18,6 +18,7 @@
 #include <boost/regex.hpp>
 #include <cctype>
 #include "velox/functions/Macros.h"
+#include "velox/functions/lib/string/StringImpl.h"
```
Comment (mbasmanova) [inline velox/functions/prestosql/URLFunctions.h:21]: Let's remove this include and move urlEscape and urlUnescape to this file.

File: velox/functions/prestosql/tests/BinaryFunctionsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+#include <array>
+#include "velox/common/base/VeloxException.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/lib/StringEncodingUtils.h"
+#include "velox/functions/lib/string/StringImpl.h"
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/BinaryFunctionsTest.cpp:21]: Do we need this include?

File: velox/functions/prestosql/tests/BinaryFunctionsTest.cpp [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/BinaryFunctionsTest.cpp:24]: Put the test into anonymous namespace inside facebook::velox::functions

**Review by mbasmanova — COMMENTED** [review]
@isadikov This is a nice improvement. Some comments.

File: velox/functions/prestosql/StringFunctions.h (hunk 1 above)
Comment (isadikov) [reply velox/functions/prestosql/StringFunctions.h:57]: Updated.

File: velox/functions/prestosql/URLFunctions.h (hunk 2 above)
Comment (isadikov) [reply velox/functions/prestosql/URLFunctions.h:21]: Updated.

File: velox/functions/prestosql/tests/BinaryFunctionsTest.cpp (hunk 4 above)
Comment (isadikov) [reply velox/functions/prestosql/tests/BinaryFunctionsTest.cpp:23]: Updated.

**Comment by isadikov** [comment]
@mbasmanova Could you take another look? I have addressed your comments and also inlined some of the functions from stringImpl into binary functions such as fromBase64, toBase64, md5, sha1, sha256, sha512, etc.

**Comment by mbasmanova** [comment]
@isadikov Is this still a draft? If not, please, update.

**Comment by isadikov** [comment]
Moved to Review mode, thanks.

**Pull request closed without merging**